packages/legend-query-builder/src/graph-manager/protocol/pure/v1/V1_QueryValueSpecificationBuilderHelper.ts

```diff
@@ -205,21 +302,40 @@ const buildAggregationExpression = (
     V1_Lambda,
     `Can't build agg() expression: agg() expects argument #1 as a lambda`,
   );
-  const processedColumnLambda =
-    returnUndefOnError(() =>
-      buildProjectionColumnLambda(
-        columnLambda,
-        openVariables,
-        compileContext,
-        processingContext,
-      ),
-    ) ??
-    new INTERNAL__UnknownValueSpecification(
-      V1_serializeValueSpecification(
-        columnLambda,
-        compileContext.extensions.plugins,
-      ),
-    );
+  let processedColumnLambda;
+  if (columnLambda.body[0] instanceof V1_AppliedFunction) {
+    processedColumnLambda =
```

Comment:
this is a little presumptious, previously, we have the check

```ts
if (
    currentPropertyExpression instanceof V1_AppliedFunction &&
    matchFunctionName(
      currentPropertyExpression.function,
      Object.values(QUERY_BUILDER_SUPPORTED_CALENDAR_AGGREGATION_FUNCTIONS),
    )
  )
```

let's add it here to be super sure this is the calendar flow and not something else we haven't supported.